### ISSUE TITLE
WPT test fails if I enforce name prohibited

### ISSUE THREAD

**Issue by aleventhal** [w3c/accname#240]
I'm working to update the Blink accessibility code to not expose a name on roles where the name is prohibited. There is a good reason for these rules, as screen readers cannot always cope with having text both in a label and in descendant content.
(Note: I want to experiment with moving the prohibited name to a description, and logging an a11y error in developer tools

However, this causes errors in external/wpt/accname/name/comp_label.html, which seem to assume that aria-label will produce an accname on these prohibited roles. Shouldn't the test be using valid ARIA?

---- Output of test ----

This is a testharness.js-based test.
Found 5 FAIL, 0 TIMEOUT, 0 NOTRUN.
[FAIL] label valid on div with associationlist role
assert_equals: <div role="associationlist" aria-label="label" data-expectedlabel="label" data-testname="label valid on div with associationlist role" class="ex">x</div> expected "label" but got ""
[FAIL] label valid on div with associationlistitemkey role
assert_equals: <div role="associationlistitemkey" aria-label="label" data-expectedlabel="label" data-testname="label valid on div with associationlistitemkey role" class="ex">x</div> expected "label" but got ""
[FAIL] label valid on div with associationlistitemvalue role
assert_equals: <div role="associationlistitemvalue" aria-label="label" data-expectedlabel="label" data-testname="label valid on div with associationlistitemvalue role" class="ex">x</div> expected "label" but got ""
[FAIL] label valid on dd element
assert_equals: <dd aria-label="label" data-expectedlabel="label" data-testname="label valid on dd element" class="ex">x</dd> expected "label" but got ""
[FAIL] label valid on dt element
assert_equals: <dt aria-label="label" data-expectedlabel="label" data-testname="label valid on dt element" class="ex">x</dt> expected "label" but got ""
Harness: the test ran to completion.

<br class="Apple-interchange-newline">

**Comment by aleventhal** [comment]
Related: issue 241, which is for ACCNAME testable statements.

**Comment by aleventhal** [comment]
This WPT test currently prevents Chrome from experimentally repairing prohibited names in content, e.g. by moving the name to the description of the element.

@accdc @cookiecrook @scottaohara WDYT?

**Comment by scottaohara** [comment]
maybe any tests for prohibited name elements/roles should be tentative?  i mean, one of the reasons naming is prohibited is based on the fact that the support was spotty / the fact that screen readers behaved differently even if the element did expose a name.

but for these elements/roles specifically, these roles are in flux / not actually in the aria spec.  so maybe these specific tests need to be rethought?

**Comment by cookiecrook** [comment]
If I understand correctly, Aaron is suggesting that the current label test is invalid because it prevents adding prohibited name tests... If that's correct Aaron, then I support removing specific label tests for elements or roles with name prohibited. If so, this issue should be moved to `web-platform-tests/interop-accessibility`, rather than `w3c/accname` @spectranaut or @jnurthen may be able to move it.

**Comment by cookiecrook** [comment]
Referenced subtests starting here: 
https://github.com/web-platform-tests/wpt/blob/master/accname/name/comp_label.html#L28

**Comment by aleventhal** [comment]
@cookiecrook Ok. Let's do this. I'll create a prototype of my idea so that we can play with it and then decide whether to standardize it. 

Assumption: prohibited name cases cause text to variably be announced by some screen readers and not others
Mitigations: 
1. User agent may move the accessible name to the description
2. If user agent does #1, it should also log an an error to the developer console or otherwise provide the error in its developer tools, with information about how to address it.

**Comment by MelSumner** [comment]
> "I think AccName should add an exit clause for roles with name prohibited…"

I agree that we could use some more actionable guidance on what kind of outcome is desired here.

**Comment by aleventhal** [comment]
@MelSumner In the meantime I'm going to build some experiments in Chrome so that you can pass a command line argument to enable different behavior, and we can try some things.

**Comment by spectranaut** [comment]
Discussed in today's ARIA meeting: https://www.w3.org/2024/07/11-aria-minutes.html#t07

**Comment by cookiecrook** [comment]
@aleventhal can you help me recall if these WPT tests need to be removed?
- https://github.com/web-platform-tests/interop-accessibility/issues/133

**Comment by zcorpan** [comment]
>  Shouldn't the test be using valid ARIA?

Tests should exercise UA requirements and verify that the requirements are followed, irrespective from any author requirements.

In https://w3c.github.io/accname/#computation-steps I see 

> If the root node's role prohibits naming, return the empty string ("").

**Comment by aleventhal** [comment]
Thanks @zcorpan. AIUI that means the test actually have the wrong expectations.
E.g.

```
expected "label" but got ""
[FAIL] label valid on dd element
```

**Comment by MelSumner** [comment]
Could I get someone to explain how failing WPT tests are preventing experimentation? 

I think I might be missing a beat here, apologies.

**Comment by cookiecrook** [comment]
@MelSumner Generally the automation commit queues of the browser engines have a CI check that expects no new failures in either the engine-internal automation, or the shared WPT automation. Since Aaron's change triggers a new failure in this (now challenged) test expectation, the solution would be to either: 

1. Change the patch so that the new failure no longer fails
2. Remove the failing test if it's determined to be an invalid expectation.
3. Add a test failure exception expectation to the patch. (Different implementations have different rules about when this patch is okay to use, but I don't think it'd be expected in this particular case.)

**Comment by cookiecrook** [comment]
@aardrian wrote:
> AIUI that means the test actually have the wrong expectations.

Okay, @aleventhal please review the WPT PR. Thanks.
- https://github.com/web-platform-tests/wpt/pull/47608

**Comment by cookiecrook** [comment]
I removed that test rather than changing it. I could not change the expectation to an empty string, since WebKit and Gecko are not disallowed from passing the label through, but Chromium also isn't required to pass the authoring error provided label though to the user trying to access their 401k. 😉

**Comment by aardrian** [comment]
@cookiecrook 

> @aardrian wrote:

That was Aaron, but I am flattered you think I am that hip with my initialisms.
